<a href="https://colab.research.google.com/github/kaindoh/Ecommerce-Customer-Service/blob/master/Ecommerce_Customers.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Loading the Libraries

In [0]:
install.packages("DataExplorer")

In [0]:
library(DataExplorer)

In [0]:
# Importing libaries

library(tidyverse)

In [0]:
library(ggplot2)

# Loading The Online Shopping Dataset

In [0]:
# Loading the dataset and displaying the first 6 rows

df = read.csv("online_shoppers_intention.csv")
head(df) 

Administrative Administrative_Duration Informational Informational_Duration
1 0               0                      0              0                    
2 0               0                      0              0                    
3 0              -1                      0             -1                    
4 0               0                      0              0                    
5 0               0                      0              0                    
6 0               0                      0              0                    
  ProductRelated ProductRelated_Duration BounceRates ExitRates PageValues
1  1               0.000000              0.20000000  0.2000000 0         
2  2              64.000000              0.00000000  0.1000000 0         
3  1              -1.000000              0.20000000  0.2000000 0         
4  2               2.666667              0.05000000  0.1400000 0         
5 10             627.500000              0.02000000  0.0500000 0         
6 19             154.216667              0.01578947  0.0245614 0         
  SpecialDay Month OperatingSystems Browser Region TrafficType
1 0          Feb   1                1       1      1          
2 0          Feb   2                2       1      2          
3 0          Feb   4                1       9      3          
4 0          Feb   3                2       2      4          
5 0          Feb   3                3       1      4          
6 0          Feb   2                2       1      3          
  VisitorType       Weekend Revenue
1 Returning_Visitor FALSE   FALSE  
2 Returning_Visitor FALSE   FALSE  
3 Returning_Visitor FALSE   FALSE  
4 Returning_Visitor FALSE   FALSE  
5 Returning_Visitor  TRUE   FALSE  
6 Returning_Visitor FALSE   FALSE

# Getting the information of the Dataset

In [0]:
# Getting the column names

attributes(df)$names

[1] "Administrative"          "Administrative_Duration"
 [3] "Informational"           "Informational_Duration" 
 [5] "ProductRelated"          "ProductRelated_Duration"
 [7] "BounceRates"             "ExitRates"              
 [9] "PageValues"              "SpecialDay"             
[11] "Month"                   "OperatingSystems"       
[13] "Browser"                 "Region"                 
[15] "TrafficType"             "VisitorType"            
[17] "Weekend"                 "Revenue"

In [0]:
# Getting the class ofthe data

attributes(df)$class

[1] "data.frame"

In [0]:
# Getting the data type of the individual column

glimpse(df)

Observations: 12,330
Variables: 18
$ Administrative          <int> 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0...
$ Administrative_Duration <dbl> 0, 0, -1, 0, 0, 0, -1, -1, 0, 0, 0, 0, 0, 0...
$ Informational           <int> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0...
$ Informational_Duration  <dbl> 0, 0, -1, 0, 0, 0, -1, -1, 0, 0, 0, 0, 0, 0...
$ ProductRelated          <int> 1, 2, 1, 2, 10, 19, 1, 1, 2, 3, 3, 16, 7, 6...
$ ProductRelated_Duration <dbl> 0.000000, 64.000000, -1.000000, 2.666667, 6...
$ BounceRates             <dbl> 0.200000000, 0.000000000, 0.200000000, 0.05...
$ ExitRates               <dbl> 0.200000000, 0.100000000, 0.200000000, 0.14...
$ PageValues              <dbl> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0...
$ SpecialDay              <dbl> 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.4, 0.0, 0.8...
$ Month                   <fct> Feb, Feb, Feb, Feb, Feb, Feb, Feb, Feb, Feb...
$ OperatingSystems        <int> 1, 2, 4, 3, 3, 2, 2, 1, 2, 2, 1, 1, 1, 2, 3...
$ Browser        

In [0]:
# Checking the unique values of columns that are factors 

fac.columns <- c('Month','VisitorType')
no <- function(x) {  
    levels(x)
  }
  
number <- sapply(select(df, fac.columns), no)
number

$Month
 [1] "Aug"  "Dec"  "Feb"  "Jul"  "June" "Mar"  "May"  "Nov"  "Oct"  "Sep" 

$VisitorType
[1] "New_Visitor"       "Other"             "Returning_Visitor"

The month of April and January are not there

In [0]:
# Checking the unique values of columns that are numerical 

num.columns <- c('Administrative','Informational','SpecialDay','OperatingSystems','Browser','Region','TrafficType','Weekend','Revenue')
no <- function(x) {  
    unique(x)
  }
  
number <- sapply(select(df, num.columns), no)
number

$Administrative
 [1]  0  1  2  4 12  3 10  6  5  9  8 16 13 11  7 18 14 17 19 15 NA 24 22 21 20
[26] 23 27 26

$Informational
 [1]  0  1  2  4 16  5  3 14  6 12  7 NA  9 10  8 11 24 13

$SpecialDay
[1] 0.0 0.4 0.8 1.0 0.2 0.6

$OperatingSystems
[1] 1 2 4 3 7 6 8 5

$Browser
 [1]  1  2  3  4  5  6  7 10  8  9 12 13 11

$Region
[1] 1 9 2 3 4 5 6 7 8

$TrafficType
 [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 18 19 16 17 20

$Weekend
[1] FALSE  TRUE

$Revenue
[1] FALSE  TRUE

In [0]:
# This the shape of the dataset

dim(df)

[1] 12330    18

we have 12330 rows and 18 columns

In [0]:
# Summary of numeric columns

df %>%
  select_if(is.numeric) %>%
  map(~summary(.))

$Administrative
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  0.000   0.000   1.000   2.318   4.000  27.000      14 

$Administrative_Duration
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  -1.00    0.00    8.00   80.91   93.50 3398.75      14 

$Informational
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  0.000   0.000   0.000   0.504   0.000  24.000      14 

$Informational_Duration
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
  -1.00    0.00    0.00   34.51    0.00 2549.38      14 

$ProductRelated
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
   0.00    7.00   18.00   31.76   38.00  705.00      14 

$ProductRelated_Duration
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
   -1.0   185.0   599.8  1196.0  1466.5 63973.5      14 

$BounceRates
    Min.  1st Qu.   Median     Mean  3rd Qu.     Max.     NA's 
0.000000 0.000000 0.003119 0.022152 0.016684 0.200000       14 

$ExitRates
   Min. 1st Qu.  Median    Mean 3rd 

# Data Cleaning

In [0]:
# Checking for missing values
colSums(is.na(df))

Administrative Administrative_Duration           Informational 
                     14                      14                      14 
 Informational_Duration          ProductRelated ProductRelated_Duration 
                     14                      14                      14 
            BounceRates               ExitRates              PageValues 
                     14                      14                       0 
             SpecialDay                   Month        OperatingSystems 
                      0                       0                       0 
                Browser                  Region             TrafficType 
                      0                       0                       0 
            VisitorType                 Weekend                 Revenue 
                      0                       0                       0

In [0]:
# Dealing with missing values

df = na.omit(df)
print("The shape of the dataset is : ")
print(dim(df))
cat("\n")
print("There are no missing vallues")
colSums(is.na(df))

[1] "The shape of the dataset is : "
[1] 12316    18

[1] "There are no missing vallues"


Administrative Administrative_Duration           Informational 
                      0                       0                       0 
 Informational_Duration          ProductRelated ProductRelated_Duration 
                      0                       0                       0 
            BounceRates               ExitRates              PageValues 
                      0                       0                       0 
             SpecialDay                   Month        OperatingSystems 
                      0                       0                       0 
                Browser                  Region             TrafficType 
                      0                       0                       0 
            VisitorType                 Weekend                 Revenue 
                      0                       0                       0

The shape of daraset did not change as muchafter dropping null values becauses it moves from 12330 to 12316    

In [0]:
# Checking for duplicates

anyDuplicated(df)

[1] 159

In [0]:
# Dropping the duplicates

df <- df[!duplicated(df),]
anyDuplicated(df)

[1] 0

Duplicates have been dropped from 159 duplicates to 0 duplicates

In [0]:
# Checking for outliers
